### ISSUE TITLE
Problems with customizing apk file names

### ISSUE THREAD

**Issue by yohom** [flutter/flutter#22027]
When I tried to modify the file name of the release version apk through the gradle script, the build failed. This is what I add in build.gradle:
```groovy
    applicationVariants.all { variant ->
        variant.outputs.all { output ->
            def outputFile = output.outputFile
            if (outputFile != null
                    && outputFile.name.endsWith('.apk')
                    && outputFile.name.contains('-release')) {
                outputFileName = "industry_${defaultConfig.versionCode}_v${defaultConfig.versionName}_${new Date().format("yyyyMMdd")}.apk"
            }
        }
    }
```
![screen shot 2018-09-19 at 18 11 13](https://user-images.githubusercontent.com/10418364/45747092-6ba50480-bc37-11e8-8c86-ce8473d64ed0.png)

flutter doctor:
```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel beta, v0.8.2, on Mac OS X 10.13.6 17G65, locale en-CN)
[✓] Android toolchain - develop for Android devices (Android SDK 27.0.3)
[✓] iOS toolchain - develop for iOS devices (Xcode 10.0)
[✓] Android Studio (version 3.1)
[!] IntelliJ IDEA Ultimate Edition (version 2018.2.1)
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
[✓] VS Code (version 1.27.1)
[✓] Connected devices (2 available)
```

**Comment by orteney** [comment]
I have same problem when setting apk name with:
```
defaultConfig {
    ...
    setArchivesBaseName("appname-$versionName($versionCode)")
}

```
![image](https://user-images.githubusercontent.com/26565071/51593740-7a713000-1f25-11e9-85db-1247894b682d.png)
But the apk is generated and launched on the device.

**Comment by a396901990** [comment]
same issue

**Comment by zoechi** [comment]
@a396901990 
Using "add reaction" on the initial comment would increase priority while `+1` comments are rather pointless and cause lots of people to get notified for no reason.
To get notified yourself use the [Subscribe] button to the top right instead.

**Comment by kangwang1988** [comment]
I think `flutter run --flavor app1` can do the job now.

**Comment by amake** [comment]
I think this is basically a duplicate of #11866, closed as moot in favor of #13977.

**Comment by yusheng-jia** [comment]
Is there and solutions?
